components/omega/doc/design/EOS.md

```diff
@@ -0,0 +1,181 @@
+
+### 2.3 Requirement: computational efficiency
+As one of the most expensive kernels, the EOS must be as efficient as possible. Developers should be discouraged from calling the EOS too frequently. The fraction of compute time used be the EOS within Omega1 will be compared to fraction used in MPAS-Ocean. 
+
+### 2.4 Requirement: abide by license for the GSW Oceanographic Toolbox
+The GSW toolbox can only be used without modification, so a Omega-specific port of certain functions will be required to support Kokkos/GPU enabled computation. The toolbox without modifications can still be used for verification testing against the Omega implementation.
+
+### 2.5 Requirement: verification
+The Omega implementation should be compared with the test value published in Roquet et al 2015, and the official GSW Oceanographic Toolbox (supported by TEOS-10) to verify correctness.
+
+### 2.6 Requirement: check value range
+The EOS code should include a check on the range of ocean properties to assess use of TEOS-10 (equivalent to the ``ocean funnel'' check), and a truncation of state variables to use the polynomial only in the valid range (similar to the truncation in MPAS-Ocean). 
+
+### 2.7 Requirement: provide adiabatically-displaced density
+The EOS code should be able to be used to calculate density adiabatically displaced to i) the layer above, ii) the surface. This is needed to calculate the Brunt-Vaisala and for mixed-layer parameterizations. This operation should be done as efficiently as possible (e.g. can we reuse parts of the polynomial? this may also be a candidate for Kernel Fusion?).
+
+### 2.8 Desired: freezing temperature calculation
+Later versions should include calculating the freezing temperature of seawater.
+
+### 2.9 Desired: output in situ temperature
+Later versions should include calculating the in situ  temperature needed for coupling (at surface including at non-zero pressure). 
+
+### 2.10 Desired: output first derivatives
+The EOS will need to be able to produce alpha and beta (drho/dT and drho/dS) for linear expansions in the higher-order pressure gradient and for some mixing parameterizations.
+
+### 2.11 Desired: variable conversion
+Later versions should include functions to convert between conservative and potential temperatures, and absolute and practical salinity. These will be used to convert the initial conditions or compare to ocean states from other sources (MPAS-Ocean, reanalysis, other models etc.) These functions may be used offline in pre- or post-processing but need to be consistent with the EOS implementation. 
+
+### 2.12 Desired: enthalpy conversion
+Later versions should include functions to calculate the layer potential enthalpy from enthalpy fluxes to convert coupling fluxes to changes in conservative temperature in a rigorous manner. 
+
+## 3 Algorithmic Formulation
```

Comment:
can we add a quick statement about algorithmic formulations aren't needed as they are described by ... (TEOS-10 doc)?